### PR TITLE
Add no_std support

### PR DESCRIPTION
**Pull request by Zireael-N**
I also replaced `rand` with a smaller dependency, `getrandom`, because that's pretty much what `OsRng::fill_bytes` is: https://github.com/rust-random/rand/blob/bf8b5a98ac95da19953f0b3b0f6da6ec8eda940b/rand_core/src/os.rs#L66-L69.

`OsRng::fill_bytes`'s return value was previously ignored, I assume that's not intended, so I changed that.

### CODE REVIEW

**Comment by Zireael-N** [comment]
When building with `--no-default-features --features alloc` minimum supported Rust version is a bit higher, seems to be 1.36.0.

**Comment by Keats** [comment]
Can you add it to CI? The MSRV bump is fine

**Comment by Zireael-N** [comment]
Done.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,7 +31,18 @@ jobs:
...
+
+    - name: Run tests using no_std
+      if: matrix.test_no_std == true
+      run: |
+        cargo test --no-default-features --features alloc
+
+    - name: Check formatting
+      if: matrix.rust == 'stable'
+      run: |
+        cargo fmt --all -- --check
```
Comment (Keats) [inline .github/workflows/ci.yml:48]: Note to myself: I'll move it to another CI check, along with clippy probably

Comment (Zireael-N) [reply .github/workflows/ci.yml:48]: `ripgrep` actually uses a separate job: https://github.com/BurntSushi/ripgrep/blob/a2e6aec7a4d9382941932245e8854f0ae5703a5e/.github/workflows/ci.yml#L185-L200

Comment (Keats) [reply .github/workflows/ci.yml:48]: Yep, I have something like that for the Rust repos at work: https://github.com/onecodex/needletail/blob/master/.github/workflows/tests.yml

**Comment by Keats** [comment]
Thanks!

**Pull request merged**